main.go

```diff
@@ -25,7 +53,117 @@ var pongMessage = []byte(`{
 	]
 }`)
 
+type State struct {
+	db *sql.DB
+}
+
+var s *State
+
+func NewState(path string) (*State, error) {
+
+	db, err := sql.Open("postgres", path)
+	if err != nil {
+		return nil, err
+	}
+
+	err = db.Ping()
+	if err != nil {
+		return nil, err
+	}
+
+	s := &State{
+		db: db,
+	}
+	return s, nil
+}
+
+func cleanMessage(message []byte) *Msg {
+
+	var msg map[string]interface{}
+
+	if err := json.Unmarshal(message, &msg); err != nil {
+		log.Println(err)
+	}
+
+	m := &Msg{
+		msg: msg,
+	}
+
+	return m
+}
+
+func (m Msg) decodeMsg() Block {
+
+	// FORMAT of msg
+	// {
+	// 	"emit": [
+	// 	   "..",
+	// 	   {
+	// 		   "block": {
+	//				number : xxxxx
+	//				hash : xxxxx
+	//				...
+	//			}
+	// 	   }
+	// 	]
+	// }
+
+	decodedMsg := (m.msg["emit"]).([]interface{})[1].(map[string]interface{})["block"]
+
+	out, _ := json.Marshal(decodedMsg)
+
+	var rawBlock Block
+
+	if err := json.Unmarshal(out, &rawBlock); err != nil {
+		panic(err)
+	}
+
+	return rawBlock
+}
+
+type Msg struct {
+	msg map[string]interface{}
+}
+
+func (s *State) WriteBlock(block Block, table string) {
+
+	block_number := int(block.Number)
+	block_hash := block.Hash
```

Comment:
Just write down `block.Hash` in the Exec function instead of creating a new name here.

Reply:
The same applies for all the others.